src/map.c

```diff
@@ -140,14 +134,17 @@ SEXP map2_impl(SEXP env, SEXP x_name_, SEXP y_name_, SEXP f_name_, SEXP type_, S
   return out;
 }
 
-SEXP pmap_impl(SEXP env, SEXP l_name_, SEXP f_name_, SEXP type_, SEXP progress) {
-  const char* l_name = CHAR(Rf_asChar(l_name_));
-  SEXP l = Rf_install(l_name);
+SEXP pmap_impl(SEXP env, SEXP type_, SEXP progress, SEXP error_call) {
+  SEXP l = Rf_install(".l");
   SEXP l_val = PROTECT(Rf_eval(l, env));
   SEXPTYPE type = Rf_str2type(CHAR(Rf_asChar(type_)));
 
   if (!Rf_isVectorList(l_val)) {
```

Comment:
```suggestion
  if (TYPEOF(l_val) != VECSXP) {
```

`Rf_isVectorList()` allows `EXPRSXP` for some reason and I feel like we don't want that. Does that feel right @lionel- ? https://github.com/wch/r-source/blob/79298c499218846d14500255efd622b5021c10ec/src/include/Rinlinedfuns.h#L851